tools/Unix.mk

```diff
@@ -618,6 +619,40 @@ bootloader:
 clean_bootloader:
 	$(Q) $(MAKE) -C $(ARCH_SRC) clean_bootloader
 
+checkpython3:
+	@if [ -z "$$(which python3)" ]; then \
+		echo "ERROR: python3 not found in PATH"; \
+		echo "       Please install python3 or fix the PATH"; \
+		exit 1; \
+	fi
+
+# debug_info target flags to get diagnostic info without building nxdiag application
+
+SYSINFO_PARSE_FLAGS = "$(realpath $(TOPDIR))"
+SYSINFO_PARSE_FLAGS += "-fsysinfo.h"
+
+SYSINFO_FLAGS = "-c"
+SYSINFO_FLAGS += "-p"
+SYSINFO_FLAGS += -f \""$(shell echo '$(CFLAGS)' | sed 's/"/\\\\\\"/g')"\"
+SYSINFO_FLAGS += \""$(shell echo '$(CXXFLAGS)' | sed 's/"/\\\\\\"/g')"\"
+SYSINFO_FLAGS += \""$(shell echo '$(LDFLAGS)' | sed 's/"/\\\\\\"/g')"\"
+
+ifneq ($(findstring esp,$(CONFIG_ARCH_CHIP)),)
+ARCH_ESP_HALDIR = $(TOPDIR)$(DELIM)arch$(DELIM)$(CONFIG_ARCH)$(DELIM)src$(DELIM)chip$(DELIM)esp-hal-3rdparty
```

Comment:
but parse_sysinfo.py doesn't use many of these options

Reply:
Yes, `parse_sysinfo.py` uses `SYSINFO_PARSE_FLAGS` flags. Others are for `host_sysinfo.py` which are taken from  `nxdiag` example.

Reply:
so, should we just keep the flag required by parse_sysinfo.py, but move the host_sysinfo.py's options to apps/system/nxdiag.
BTW, host_sysinfo.py should be still keep in apps/system/nxdiag too.

Reply:
We can move `host_sysinfo.py` to its original place and then put related flags in `apps/system/nxdiag/makefile` but in the end, we will revert it into first version of this PR(mostly). Final decision is up to you. Important point is, if we move it we will call something in appsdir from nuttx (here/ tools/Unix.mk). Let me create an example to show it better:

```
# tools/Unix.mk
FLAGS = $(APPDIR)/system/nxdiag/sysinfo.h
FLAGS += ...
debug_info: checkpython3
ifneq ($(APPDIR),)
	$(Q) $(MAKE) -C $(APPDIR) debug_info
endif
python3 tools/parse_sysinfo.py FLAGS

# apps/Makefile:
debug_info:
$(Q) $(MAKE) -C $(APPDIR)/system/nxdiag debug_info

# apps/system/nxdiag/Makefile:
FLAGS = ...
debug_info: checkpython3
python3 $(APPDIR)/tools/host_sysinfo.py FLAGS> sysinfo.h
```